local-cluster/tests/local_cluster.rs

```diff
@@ -4480,6 +4480,239 @@ fn test_slot_hash_expiry() {
+    // minority fork)
+    let mut validator_configs =
+        make_identical_validator_configs(&ValidatorConfig::default_for_test(), node_stakes.len());
+    validator_configs[1].voting_disabled = true;
+
+    let mut config = ClusterConfig {
+        cluster_lamports: DEFAULT_CLUSTER_LAMPORTS + node_stakes.iter().sum::<u64>(),
+        node_stakes,
+        validator_configs,
+        validator_keys: Some(validator_keys),
+        node_vote_keys: Some(node_vote_keys),
+        slots_per_epoch,
+        stakers_slot_offset: slots_per_epoch,
+        skip_warmup_slots: true,
+        ..ClusterConfig::default()
+    };
+    let mut cluster = LocalCluster::new(&mut config, SocketAddrSpace::Unspecified);
+
+    let mut common_ancestor_slot = 8;
+
+    let a_ledger_path = cluster.ledger_path(&a_pubkey);
+    let b_ledger_path = cluster.ledger_path(&b_pubkey);
+
+    // Immediately kill B and C (we just needed it for the initial stake distribution)
+    info!("Killing B and C");
+    let mut b_info = cluster.exit_node(&b_pubkey);
+    cluster.exit_node(&c_pubkey);
+
+    // Let A run for a while until we get to the common ancestor
+    info!("Letting A run until common_ancestor_slot");
+    loop {
+        if let Some((last_vote, _)) = last_vote_in_tower(&a_ledger_path, &a_pubkey) {
+            if last_vote >= common_ancestor_slot {
+                break;
+            }
+        }
+        sleep(Duration::from_millis(100));
+    }
+
+    // Keep A running, but setup B so that it thinks it has voted up until common ancestor (but
+    // doesn't know anything past that)
+    {
+        info!("Copying A's ledger to B");
+        std::fs::remove_dir_all(&b_info.info.ledger_path).unwrap();
+        let mut opt = fs_extra::dir::CopyOptions::new();
+        opt.copy_inside = true;
+        fs_extra::dir::copy(&a_ledger_path, &b_ledger_path, &opt).unwrap();
+
+        // remove A's tower in B's new copied ledger
+        info!("Removing A's tower in B's ledger dir");
+        remove_tower(&b_ledger_path, &a_pubkey);
+
+        // load A's tower and save it as B's tower
+        info!("Loading A's tower");
+        if let Some(mut a_tower) = restore_tower(&a_ledger_path, &a_pubkey) {
+            a_tower.node_pubkey = b_pubkey;
+            // Update common_ancestor_slot because A is still running
+            if let Some(s) = a_tower.last_voted_slot() {
+                common_ancestor_slot = s;
+                info!("New common_ancestor_slot {}", common_ancestor_slot);
+            } else {
+                panic!("A's tower has no votes");
+            }
+            info!("Increase lockout by 6 confirmation levels and save as B's tower");
+            a_tower.increase_lockout(6);
+            save_tower(&b_ledger_path, &a_tower, &b_info.info.keypair);
+            info!("B's new tower: {:?}", a_tower.tower_slots());
+        } else {
+            panic!("A's tower is missing");
+        }
```

Comment:
Seems like we could remove all of this lockout setup logic and tower manipulation if we:

Create `A`, `B`, `C` to have total < SWITCHING_THRESHOLD of the stake, so nobody can switch. Create a custom leader schedule where `A`, `B`, and `C` have all the leader slots so that the test can proceed faster.

1. Let `A` and `B` and `C` run for `X` slots
2. Turn off `B`
3. Let `A` make a fork and vote on its fork
4. Turn off `A`
5. Turn on `B`, it won't repair `A`'s fork so it should make its own fork and vote on its own fork. 
6. Turn off `B`. Copy `B`'s ledger to `C`, let `C` create blockhash expiration number of blocks *past* B's last vote on its own fork
8. Copy both `A` and  `C`'s ledger back to `B`. `B` should try to switch to `A`, fail, and then refresh its vote to the tip of `C`'s fork

Reply:
Sorry I misspoke, we shoudl wait for `B` to create a Slothash expiration number of blocks, which is twice as many as the blockhash expiration number of blocks.

I think the problem if you don't have C  then this part "let B create blockhash expiration number of blocks on its own fork.", means that `B` will try to refresh its vote *sooner* after the blockhash expires instead of after the SlotHash expiration number of blocks

Reply:
Actually in this test, because I shrunk slothash size to 8, slothash expiration happened before blockhash expiration, so I never saw vote refresh caused by blockhash expiration in the test.

Reply:
oh, where do we configure the SlotHash size? I didn't know we could do that

Reply:
solana_sdk::slot_hashes::set_entries_for_tests_only(8);
This is for test only.